src/crossql/Migrations/MigrationRunner.cs

```diff
@@ -20,182 +19,230 @@ public MigrationRunner(IDbProvider dbProvider)
+                switch (migrationStep)
+                {
+                    case MigrationStep.Setup:
+                        failedStep = MigrationStep.Setup;
+                        await migration.SetupMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.Migrate:
+                        failedStep = MigrationStep.Migrate;
+                        await migration.Migrate(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.Finish:
+                        failedStep = MigrationStep.Finish;
+                        await migration.FinishMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ClientSetup:
+                        failedStep = MigrationStep.ClientSetup;
+                        await migration.ClientSetupMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ClientMigrate:
+                        failedStep = MigrationStep.ClientMigrate;
+                        await migration.ClientMigrate(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ClientFinish:
+                        failedStep = MigrationStep.ClientFinish;
+                        await migration.ClientFinishMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ServerSetup:
+                        failedStep = MigrationStep.ServerSetup;
+                        await migration.ServerSetupMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ServerMigrate:
+                        failedStep = MigrationStep.ServerMigrate;
+                        await migration.ServerMigrate(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ServerFinish:
+                        failedStep = MigrationStep.ServerFinish;
+                        await migration.ServerFinishMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(migrationStep), migrationStep, $"An invalid {nameof(MigrationStep)} was passed to the {nameof(MigrationRunner)}");
+                }
 
-        private async Task<DatabaseVersionModel> GetMigrationInformationAsync(IDbMigration migration)
+                var dbCommand = database.ToString();
+                if (!string.IsNullOrWhiteSpace(dbCommand))
+                    await _dbProvider.ExecuteNonQuery(dbCommand);
+            }
+            catch (Exception ex)
+            {
+                var database = new Database(dbProvider.DatabaseName, dbProvider.Dialect);
```

Comment:
If any one of the migrations fail, we run the failed migration method and exit. Is this the best way, or should subsequent migrations still continue?

_I feel as though they should not_